FirebaseStorage/Tests/SwiftIntegration/StorageIntegration.swift

```diff
@@ -0,0 +1,672 @@
+    let ref = storage?.reference(withPath: "ios/public/-._~!$'()*,=:@&+;")
+    let data = try XCTUnwrap("Hello Swift World".data(using: .utf8), "Data construction failed")
+    ref?.putData(data, metadata: nil, completion: { metadata, error in
+      XCTAssertNotNil(metadata, "Metadata should not be nil")
+      XCTAssertNil(error, "Error should be nil")
+      expectation.fulfill()
+    })
+    waitForExpectations()
+  }
+
+  func testUnauthenticatedSimplePutDataInBackgroundQueue() throws {
+    let expectation = self.expectation(description: "testUnauthenticatedSimplePutDataInBackgroundQueue")
+    let ref = storage?.reference(withPath: "ios/public/testBytesUpload")
+    let data = try XCTUnwrap("Hello Swift World".data(using: .utf8), "Data construction failed")
+    DispatchQueue.global(qos: .background).async {
+      ref?.putData(data, metadata: nil, completion: { metadata, error in
+        XCTAssertNotNil(metadata, "Metadata should not be nil")
+        XCTAssertNil(error, "Error should be nil")
+        expectation.fulfill()
+      })
+    }
+    waitForExpectations()
+  }
+
+  func testUnauthenticatedSimplePutEmptyData() {
+    let expectation = self.expectation(description: "testUnauthenticatedSimplePutEmptyData")
+    let ref = storage?.reference(withPath: "ios/public/testUnauthenticatedSimplePutEmptyData")
+    let data = Data()
+    ref?.putData(data, metadata: nil, completion: { metadata, error in
+      XCTAssertNotNil(metadata, "Metadata should not be nil")
+      XCTAssertNil(error, "Error should be nil")
+      expectation.fulfill()
+    })
+    waitForExpectations()
+  }
+
+  func testUnauthenticatedSimplePutDataUnauthorized() throws {
+    let expectation = self.expectation(description: "testUnauthenticatedSimplePutDataUnauthorized")
+    let ref = storage?.reference(withPath: "ios/private/secretfile.txt")
+    let data = try XCTUnwrap("Hello Swift World".data(using: .utf8), "Data construction failed")
+    ref?.putData(data, metadata: nil, completion: { metadata, error in
+      XCTAssertNil(metadata, "Metadata should be nil")
+      XCTAssertNotNil(error, "Error should not be nil")
+      XCTAssertEqual((error! as NSError).code, StorageErrorCode.unauthorized.rawValue)
+      expectation.fulfill()
+    })
+    waitForExpectations()
+  }
+
+  func testUnauthenticatedSimplePutFile() throws {
+    let expectation = self.expectation(description: "testUnauthenticatedSimplePutFile")
+    let ref = storage?.reference(withPath: "ios/public/testUnauthenticatedSimplePutFile")
+    let data = try XCTUnwrap("Hello Swift World".data(using: .utf8), "Data construction failed")
+    let tmpDirURL = URL(fileURLWithPath: NSTemporaryDirectory())
+    let fileURL = tmpDirURL.appendingPathComponent("hello.txt")
+    do {
+      try data.write(to: fileURL, options: Data.WritingOptions.atomicWrite)
+    } catch {
+      XCTFail("Write to \(fileURL) failed")
```

Comment:
Is this just so we can catch the `fileURL` here? Since the function throws it doesn't need to be in a `do/catch` block.

Reply:
Done